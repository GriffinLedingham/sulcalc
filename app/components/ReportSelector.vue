<template>
  <div class="row">
    <div role="group" class="col-6 btn-group-vertical">
      <button
        v-for="{ report, index } in reports"
        :key="index"
        type="button"
        :class="buttonClasses(index)"
        @click="selectReport({ index })"
      >
        {{ report.move.name }}
      </button>
    </div>
    <div role="group" class="col-6">
      <div v-for="{ report, index } in reports" :key="index" class="row py-2">
        {{ report.minPercent || 0 }}% - {{ report.maxPercent || 0 }}%
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters, mapActions, mapState } from "vuex";
import ButtonRadioGroup from "./ui/ButtonRadioGroup.vue";

export default {
  components: {
    ButtonRadioGroup
  },
  props: {
    reports: {
      required: true,
      type: Array
    }
  },
  computed: {
    ...mapState(["reportStick"]),
    ...mapGetters(["selectedReportIndex"])
  },
  methods: {
    ...mapActions(["selectReport"]),
    buttonClasses(reportIndex) {
      const isReportSelected = reportIndex === this.selectedReportIndex;
      const usesStuckColor = this.reportStick && isReportSelected;
      return {
        btn: true,
        "btn-primary": !usesStuckColor,
        "btn-secondary": usesStuckColor,
        active: isReportSelected
      };
    }
  }
};
</script>
