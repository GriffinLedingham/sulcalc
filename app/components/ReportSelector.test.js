import { createLocalVue, shallowMount } from "@vue/test-utils";
import { install as vuexInstall, Store } from "vuex";
import ReportSelector from "./ReportSelector.vue";

const localVue = createLocalVue();
localVue.use(vuexInstall);

const reports = [
  {
    index: 0,
    report: {
      move: { name: "Fire Blast" },
      minPercent: 91.1,
      maxPercent: 95.5
    }
  },
  {
    index: 1,
    report: {
      move: { name: "Surf" },
      minPercent: 20.1,
      maxPercent: 25.3
    }
  },
  {
    index: 2,
    report: {
      move: { name: "Fusion Bolt" }
    }
  },
  {
    index: 3,
    report: {
      move: { name: "Bounce" },
      minPercent: 30.2,
      maxPercent: 37.8
    }
  }
];

test("highlights the selected report", () => {
  const wrapper = shallowMount(ReportSelector, {
    localVue,
    store: new Store({
      state: {
        reportStick: false
      },
      getters: {
        selectedReportIndex: jest.fn().mockReturnValue(0)
      },
      actions: {
        selectReport: jest.fn()
      }
    }),
    propsData: { reports }
  });
  expect(wrapper.html()).toMatchSnapshot();
});

test("shows the report is stickied", () => {
  const wrapper = shallowMount(ReportSelector, {
    localVue,
    store: new Store({
      state: {
        reportStick: true
      },
      getters: {
        selectedReportIndex: jest.fn().mockReturnValue(1)
      },
      actions: {
        selectReport: jest.fn()
      }
    }),
    propsData: { reports }
  });
  expect(wrapper.html()).toMatchSnapshot();
});
