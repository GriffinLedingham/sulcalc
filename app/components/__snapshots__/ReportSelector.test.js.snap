// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlights the selected report 1`] = `
"<div class=\\"row\\"><div role=\\"group\\" class=\\"col-6 btn-group-vertical\\"><button type=\\"button\\" class=\\"btn btn-primary active\\">
      Fire Blast
    </button><button type=\\"button\\" class=\\"btn btn-primary\\">
      Surf
    </button><button type=\\"button\\" class=\\"btn btn-primary\\">
      Fusion Bolt
    </button><button type=\\"button\\" class=\\"btn btn-primary\\">
      Bounce
    </button></div> <div role=\\"group\\" class=\\"col-6\\"><div class=\\"row py-2\\">
      91.1% - 95.5%
    </div><div class=\\"row py-2\\">
      20.1% - 25.3%
    </div><div class=\\"row py-2\\">
      0% - 0%
    </div><div class=\\"row py-2\\">
      30.2% - 37.8%
    </div></div></div>"
`;

exports[`shows the report is stickied 1`] = `
"<div class=\\"row\\"><div role=\\"group\\" class=\\"col-6 btn-group-vertical\\"><button type=\\"button\\" class=\\"btn btn-primary\\">
      Fire Blast
    </button><button type=\\"button\\" class=\\"btn btn-secondary active\\">
      Surf
    </button><button type=\\"button\\" class=\\"btn btn-primary\\">
      Fusion Bolt
    </button><button type=\\"button\\" class=\\"btn btn-primary\\">
      Bounce
    </button></div> <div role=\\"group\\" class=\\"col-6\\"><div class=\\"row py-2\\">
      91.1% - 95.5%
    </div><div class=\\"row py-2\\">
      20.1% - 25.3%
    </div><div class=\\"row py-2\\">
      0% - 0%
    </div><div class=\\"row py-2\\">
      30.2% - 37.8%
    </div></div></div>"
`;
